### PR TITLE
Added diff functionality for Git and SVN

### PR DESCRIPTION
**Pull request by shanept**
Creates a function 'getDiff' for repositories, which returns the raw output of the diff command for the respective repository engine.

### CODE REVIEW

File: src/TQ/Git/Repository/Repository.php [hunk 1]
```diff
@@ -903,6 +903,29 @@ public function getStatus()
     }
 
     /**
+     * Returns the diff of a file
+     *
+     * @param   string  $file       The path to the file
+     * @param   bool    $staged     Should the diff return for the staged file
+     * @return  string
+     */
+    public function getDiff($file, $staged=false)
```
Comment (sgehrig) [inline src/TQ/Git/Repository/Repository.php]: Sorry for being picky ;-) Could you add some whitespace around the `=`?

```
public function getDiff($file, $staged = false)
```

**Comment by sgehrig** [comment]
Generally I'd favor a more structured return value - at least an array of diffs (one per file). But I think it's OK for now as everybody could parse the response to create a structured value.

**Comment by shanept** [comment]
I didn't expect people to search for diffs⁣ from multiple files, but now that you mention it I do see the validity of such an approach.

I will add those spaces, for sure.

File: src/TQ/Git/Repository/Repository.php (hunk 1 above)
Comment (shanept) [inline src/TQ/Git/Repository/Repository.php]: Of course!

**Comment by sgehrig** [comment]
Perfect! Thanks for your contribution.

**Comment by shanept** [comment]
More than happy to help. Thank you

**Comment by sgehrig** [comment]
Please don't forget to rebase to current master and squash all commits. THX!

**Comment by shanept** [comment]
Ok, I have modified it so you can pass an array of filenames and get an array of diffs for each file, or by not passing any filenames (or null), it will return diffs for all files currently modified.

I have also squashed all commits.

**Comment by sgehrig** [comment]
There seems to be a problem with the `TQ\Tests\Svn\Repository\InfoTest::testGetDiff` test when running on PHP 5.3 and 5.4 with PHPUnit 4.8.31 (this is the highest version available for PHP < 7) respectively. The order of the resulting array seems to be different. Rather weird from my perspective.

**Comment by shanept** [comment]
Hi Stefan,

That is peculiar. I ran it on PHP7 and a different version of PHPUnit and didn't have a problem, except with some deprecations in older versions. I'll check it out and get back with a fix.
Thanks⁣

**Comment by sgehrig** [comment]
Strange - indeed. Perhaps it's possible to adjust the test so that the order doesn't matter. 
Which component defines the order in these arrays? Is it the underlying SVN cli command? Or is it something in PHP?

**Comment by shanept** [comment]
Stefan, the order is defined by the order in which the⁣ the files are passed to the diff function. Otherwise it is defined by the order in which the getStatus call returns them. Simple fix would be to sort the array by key before comparing them

**Comment by sgehrig** [comment]
It indeed seems as if the `getStatus()` method for SVN does not guarantee a strict order - for whatever reason. So sorting might be the best option IMHO.

**Comment by shanept** [comment]
Stefan, I have added sorting to the SVN diff.

**Comment by sgehrig** [comment]
Looks good. Thanks!

**Comment by shanept** [comment]
Awesome, thank you!

**Pull request merged**